LIPS/lip-25.md

```diff
@@ -0,0 +1,1983 @@
+    uint256 stuckPenaltyEndTimestamp;
+    /// @notice The total number of validators in the EXITED state on the Consensus Layer
+    /// @dev This value can't decrease in normal conditions
+    uint256 totalExitedValidators;
+    /// @notice The total number of validators deposited via the official Deposit Contract
+    /// @dev This value is a cumulative counter: even when the validator goes into EXITED state this
+    ///     counter is not decreasing
+    uint256 totalDepositedValidators;
+    /// @notice The number of validators in the set available for deposit
+    uint256 depositableValidatorsCount;
+}
+
+/// @notice Updates the limit of the validators that can be used for deposit
+/// @param _stakingModuleId Id of the staking module
+/// @param _nodeOperatorId Id of the node operator
+/// @param _targetLimitMode Target limit mode
+/// @param _targetLimit Target limit of the node operator
+function updateTargetValidatorsLimits(
+    uint256 _stakingModuleId,
+    uint256 _nodeOperatorId,
+    uint256 _targetLimitMode,
+    uint256 _targetLimit
+) external onlyRole(STAKING_MODULE_MANAGE_ROLE) {
+    _getIStakingModuleById(_stakingModuleId).updateTargetValidatorsLimits(
+        _nodeOperatorId,
+        _targetLimitMode,
+        _targetLimit
+    );
+}
+
+/// @notice Returns node operator summary from the staking module
+/// @param _stakingModuleId id of the staking module where node operator is onboarded
+/// @param _nodeOperatorId id of the node operator to return summary for
+function getNodeOperatorSummary(
+    uint256 _stakingModuleId,
+    uint256 _nodeOperatorId
+) public view returns (NodeOperatorSummary memory summary) {
+    StakingModule memory stakingModuleState = getStakingModule(_stakingModuleId);
+    IStakingModule stakingModule = IStakingModule(stakingModuleState.stakingModuleAddress);
+    /// @dev using intermediate variables below due to "Stack too deep" error in case of
+    ///     assigning directly into the NodeOperatorSummary struct
+    (
+        uint256 targetLimitMode,
+        uint256 targetValidatorsCount,
+        uint256 stuckValidatorsCount,
+        uint256 refundedValidatorsCount,
+        uint256 stuckPenaltyEndTimestamp,
+        uint256 totalExitedValidators,
+        uint256 totalDepositedValidators,
+        uint256 depositableValidatorsCount
+    ) = stakingModule.getNodeOperatorSummary(_nodeOperatorId);
+    summary.targetLimitMode = targetLimitMode;
+    summary.targetValidatorsCount = targetValidatorsCount;
+    summary.stuckValidatorsCount = stuckValidatorsCount;
+    summary.refundedValidatorsCount = refundedValidatorsCount;
+    summary.stuckPenaltyEndTimestamp = stuckPenaltyEndTimestamp;
+    summary.totalExitedValidators = totalExitedValidators;
+    summary.totalDepositedValidators = totalDepositedValidators;
+    summary.depositableValidatorsCount = depositableValidatorsCount;
+}
+```
+
+#### 2.5. New methods for getting deposit blocks
```

Comment:
```suggestion
#### 2.5. New methods for getting module deposit limits
```

Reply:
It is important that we migrate these params from DSM to SR, since each module may have different ones

Reply:
OK, changed the title and added the relevant notes to this section.